### PR TITLE
[fc] Log exception when FC increment unhealthy request metric

### PR DESCRIPTION
**Pull request by m-nagarajan**
## Summary
Log exception with a redundant filter when fast client increments unhealthy request metric

## How was this PR tested?
GH CI

## Does this PR introduce any user-facing changes?
- [x] No. You can skip the rest of this section.
- [] Yes. Make sure to explain your proposed changes and call out the behavior change.

### CODE REVIEW

File: clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java [hunk 1]
```diff
@@ -34,6 +36,8 @@ public class StatsAvroGenericStoreClient<K, V> extends DelegatingAvroStoreClient
 
   private final int maxAllowedKeyCntInBatchGetReq;
   private final boolean useStreamingBatchGetAsDefault;
+  private final RedundantExceptionFilter redundantExceptionFilter =
+      new RedundantExceptionFilter(RedundantExceptionFilter.DEFAULT_BITSET_SIZE, TimeUnit.SECONDS.toMillis(1));
```
Comment (huangminchn) [inline clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java:40]: Can we change it to one log per minute?

File: clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java [hunk 2]
```diff
@@ -163,6 +167,13 @@ private <R> CompletableFuture<R> recordRequestMetrics(
          * considered an unhealthy request: so only incrementing a subset of metrics.
          */
         exceptionReceived = true;
+        if (!redundantExceptionFilter.isRedundantException(throwable.getMessage())) {
```
Comment (huangminchn) [inline clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java:170]: I am not sure if this message content is consistent. Can we just use the store name, or does it support passing nothing and always print one log per minute?

**Review by huangminchn — CHANGES_REQUESTED** [review]

File: clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java (hunk 1 above)
Comment (m-nagarajan) [reply clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java]: changed it to default which is one log per minute.

File: clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java (hunk 2 above)
Comment (m-nagarajan) [reply clients/venice-client/src/main/java/com/linkedin/venice/fastclient/StatsAvroGenericStoreClient.java:168]: That is a good point. To start with, we can get all unique logs per minute and once validated and fixed, we can remove this or make it a stricter one log per minute.

**Pull request closed without merging**